File: Terminplaner-Java/src/Terminkalender.java
import java.util.Collection;
import java.util.Date;

public class Terminkalender {

	private final String _besitzer;
	private final TerminRepository _repository;
	
	public Terminkalender(TerminRepository repository, String besitzer) {
		_repository = repository;
		_besitzer = besitzer;
	}
	
	public Termin newTermin(Date jetzt, int dauerInMinuten, String bezeichnung) {
		Termin t = new Termin(_besitzer, jetzt, dauerInMinuten, bezeichnung);
		_repository.speichere(t);
		return t;
	}

	boolean hatTermin(Termin termin) {
		return _repository.hatTermin(termin);
	}

	public Collection<Termin> getTermine() {
		return getTermine(False);
	}

	public Collection<Termin> getTermine(boolean zeigeAuchAbgelehnte) {
		return _repository.getTermine(_besitzer, zeigeAuchAbgelehnte);
	}

}

File: Terminplaner-Ruby/src/Terminkalender.rb
require "Termin"

class Terminkalender

	def initialize(repository, besitzer)
		@repository = repository
		@besitzer = besitzer
	end
	
	def new_termin(jetzt, dauer_in_minuten, bezeichnung) 
		termin = Termin.new(@besitzer, jetzt, dauer_in_minuten, bezeichnung)
		@repository.speichere termin
		termin
	end

	def hat_termin(termin)
		@repository.hat_termin termin 
	end

	def alle_termine
		termine false
	end
	
	def termine(zeige_auch_abgelehnte)
		@repository.termine(@besitzer, zeige_auch_abgelehnte)
	end

end

